cmd/root/root.go

```diff
@@ -169,11 +169,22 @@ Stack Trace:
 		exitCode = 1
 	}
 
+	commandStr := commandString(cmd)
 	ctx = cmd.Context()
-	telemetryErr := telemetry.Upload(ctx, protos.ExecutionContext{
+
+	// Log bundle deploy failures. Only log if we have successfully configured
+	// an authenticated Databricks client. We cannot log unauthenticated telemetry
+	// from the CLI yet.
+	if cmdctx.HasConfigUsed(ctx) && commandStr == "bundle_deploy" && exitCode != 0 {
+		telemetry.Log(ctx, protos.DatabricksCliLog{
+			BundleDeployEvent: &protos.BundleDeployEvent{},
```

Comment:
where does it record that it's a failure in the payload?

Reply:
In the exit_code. It's a part of the `execution_context` field:
```
      "execution_context": {
        "cmd_exec_id": "[CMD-EXEC-ID]",
        "version": "[DEV_VERSION]",
        "command": "bundle_deploy",
        "operating_system": "[OS]",
        "execution_time_ms": SMALL_INT,
        "exit_code": 1
      },
```